packages/ui/src/utils/urlBuilder.ts

```diff
@@ -1,12 +1,24 @@
 export const buildW3iUrl = (
   base: string,
+  dappInfo: {
+    name: string;
+    icon: string;
+    description: string;
+  },
   account?: string,
   uiEnabled?: { chat?: string; push?: string; settings?: string }
 ) => {
   const searchParams = new URLSearchParams();
   if (account) {
+    searchParams.append("authProvider", "external");
     searchParams.append("account", account);
   }
+
+  searchParams.append("dappContext", window.location.origin);
```

Comment:
Just realised that this is kind of overloading the term "context" between what it means here and normally in React.

Could we use `dappOrigin` or similar instead for this to avoid the ambiguity? Didn't notice this in the w3i PR review.